packages/utils/src/validation/ObjectValidator.js

```diff
@@ -41,12 +40,46 @@ export class ObjectValidator {
       }
     }
   }
+
+  /**
+   * @param {Object<string, any>} object          The object to be validated
+   * @param {function} [constructError]  An optional function to construct a custom error
+   */
+  validateSync(object, constructError) {
```

Comment:
I admit that this is not implemented ideally:
* We could generalise some aspects of the `validateSync` and `validate` functions
* We should probably use "sunc" validation by default, and explicitly call async validator (`validateAsync`, `validate`)

But I didn't want to expand the scope the changes here. I'll add some notes to address those issues in [1201: Generalise object validation errors](https://github.com/beyondessential/tupaia-backlog/issues/1201), since the `ObjectValidator` functionality is very useful but probably needs some changes since it has been moved out of server packages.

Reply:
Actually I'll create a new card since the one mentioned above is about a different issue

Reply:
Card: https://github.com/beyondessential/tupaia-backlog/issues/2096